src/compass/utils/stitching/stitch_burst.py

```diff
@@ -0,0 +1,404 @@
+                        help='Directory with S1-A/B bursts organized by dates')
+    parser.add_argument('-b', '--burst-list', type=str, nargs='+',
+                        default=None, dest='burst_list',
+                        help='List of burst IDs to stitch. If None, common bursts'
+                             'among all dates will be stitched (default: None')
+    parser.add_argument('-m', '--margin', type=float,
+                        default=100, dest='margin',
+                        help='Margin to apply during stitching. Same units as bursts coordinate system.'
+                             '(default: 100 m, UTM)')
+    parser.add_argument('-s', '--scratchdir', type=str, default='scratch',
+                        dest='scratch',
+                        help='Directory where to store temporary results (default: scratch)')
+    parser.add_argument('-o', '--outdir', type=str, default='outdir',
+                        dest='outdir',
+                        help='Directory path where to store stitched bursts (default: outdir)')
+    return parser.parse_args()
+
+
+def main(indir, outdir, scratchdir, margin, burst_list):
+    '''
+    Stitch S1-A/B bursts for stack processing
+
+    Parameters:
+    ----------
+    indir: str
+       File path to directory containing S1-A/B bursts
+       organized by date
+    outdir: str
+       File path to directory where to store stitched bursts
+    scratchdir: str
+       File path to directory where to store intermediate
+       results (e.g., shapefiles of burst boundary)
+    margin: float
+       Margin to apply to burst boundary while stitching.
+       Same units as bursts coordinate system
+    burst_list: list [str]
+       List of burst IDs to stitch. If not provided, common
+       bursts among all dates are identified and considered
+       for stitching
+    '''
+    info_channel = journal.info('stitch_burst.main')
+    error_channel = journal.error('stitch_burst.main')
+
+    # Start tracking time
+    info_channel.log('Start burst stitching')
+    t_start = time.time()
+
+    # Check that input directory is valid
+    if not os.path.isdir(indir):
+        err_str = f'{indir} is not a valid input directory'
+        error_channel.log(err_str)
+        raise ValueError(err_str)
+
+    # Create output and scratch dirs if not existing
+    if not os.path.exists(outdir):
+        os.makedirs(outdir, exist_ok=True)
+        helpers.check_write_dir(outdir)
+    if not os.path.exists(scratchdir):
+        os.makedirs(scratchdir, exist_ok=True)
+        helpers.check_write_dir(scratchdir)
+
+    # Collect info for stitching in all dirs in 'indir'
+    # and return a panda dataframe with info
+    data_dict = get_stitching_dict(indir)
+
+    # If stitching some bursts, prune dataframe to
+    # contains only the burst IDs to stitch
+    if burst_list is not None:
+        data_dict = prune_dataframe(data_dict,
+                                    'burst_id', burst_list)
```

Comment:
This can be removed if suggestion to filter burst IDs in `get_stitching_dict`/`get_stitching_dataframe` accepted.